pipelines/trace.go

```diff
@@ -53,6 +53,19 @@ func NewTracePipeline(c PipelineConfig) (func() error, error) {
 	}, nil
 }
 
+func (c PipelineConfig) newSampler() trace.Sampler {
+	if !c.SamplingEnabled || c.SamplingPercent == 100 {
```

Comment:
Very minor performance optimisation here, compared to using `TraceIDRatioBased` unconditionally.